src/UI/Component/Button/Factory.php

```diff
@@ -209,5 +209,55 @@ public function month($default);
 	 */
 	public function tag($label, $action);
 
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The graphical button combines the recognisability of a graphical element
+	 *     with the explicity of a text-label. It has an obtrusive presentation that
+	 *     indicate important actions on the screen and make them hard to overlook.
+	 *
+	 *   composition: >
+	 *     The graphical button consists of an icon or glyph and a (very short) text.
+	 *
+	 *   effect: >
+	 *     Both, icon/glyph and text, form an entity an can only be operated together.
+	 *     The button has a "engaged"-state: When the button is used to toggle the
```

Comment:
For toggling please use the accepted View Control Mode. 

Kitchensink clearly defines Buttons as controls that "trigger interactions that change the system’s status."
Toggling does not trigger interaction. If this is about toggling please use the respective accepted control.

Reply:
I don't think that's correct. Mode View Control switches between different aspects of some data. We don't have that aspect in the Side Bar usecase, as well as we don't have that in the Page Editor usecase.

I think this is a problem of the button definition. After all, the mode view control already uses button for that exact switching purpose, which is a violation of the rules as they currently are. The Dropdown on the other hand does not use the button as a trigger, but it arguably should do that, in general and for the Page Editor use case.

For the Mode View Control and the Section View Control we also already found the need for an engaged state of the button. So imo it becomes clearer that the button indeed can do other stuff than "trigger interactions that change the system’s status" but also may trigger actions that change view's status.

Reply:
Summoning @Amstutz.

Reply:
Opened https://github.com/ILIAS-eLearning/ILIAS/pull/897 for change in button rules.